File: server/ruby/validate_payment.rb
class ValidatePayment
  require 'khipu-api-client'

  Khipu.configure do |c|
    c.secret = File.read("../SECRET").strip
    c.receiver_id = File.read("../RECEIVER_ID").strip
    c.platform = 'demo-client'
    c.platform_version = '2.0'
    #c.debugging = true
  end

  client = Khipu::PaymentsApi.new

  response = client.payments_get(File.read("../NOTIFICATION_TOKEN").strip)

  print "PAYMENT_ID: " + response.payment_id + "\n"
  print "AMOUNT: " +  response.amount.to_s + "\n"
  print "CURRENCY: " + response.currency + "\n"
  print "STATUS: " + response.status + "\n"

end

File: server/php/validate_payment.php
<?php
require __DIR__ . '/vendor/autoload.php';

$configuration = new Khipu\Configuration();
$configuration->setSecret(trim(file_get_contents("../SECRET")));
$configuration->setReceiverId(trim(file_get_contents("../RECEIVER_ID")));
$configuration->setPlatform('demo-client', '2.0');
$notificationToken = trim(file_get_contents("../NOTIFICATION_TOKEN"));
# $configuration->setDebug(true);
$client = new Khipu\ApiClient($configuration);
$payments = new Khipu\Client\PaymentsApi($client);

try {
    $response = $payments->paymentsGet($notificationToken);

    print "PAYMENT_ID: " . $response->getPaymentId() . "\n";
    print "TRANSACTION_ID: " . $response->getTransactionId() . "\n";
    print "AMOUNT: " . $response->getAmount() . "\n";
    print "CURRENCY: " . $response->getCurrency() . "\n";
    print "STATUS: " . $response->getStatus() . "\n";

} catch (Exception $e) {
    echo $e->getMessage();
}
